book/src/volume-cloning.md

```diff
@@ -0,0 +1,45 @@
+# Volume Cloning
+
+## Status
+
+* Kubernetes 1.15: Alpha
+
+## Overview
+
+A storage provider that allows volume cloning as a create feature, may choose to implement volume cloning via a
+control-plane CSI RPC call.
+
+## Implementing Volume cloning functionality
+
+To implement volume cloning the CSI driver MUST:
+
+1. Implement checks in `CreateVolume` function implementation.
+2. Implement `CLONE_VOLUME` controller capability.
+
+The cloning feature in Kubernetes is implemented by adding a `DataSource` entry of `kind: PersistentVolumeClaim` to the csi.CreateVolumeRequest.  This presents the source volume (PersistentVolumeClaim to clone from) as a generic data source.  The source volume must:
```

Comment:
Unsure if we need to document this part about Kubernetes user requirements. There's nothing a driver author can do for this right?